### DISCUSSION TITLE
Specifying virtualenv script path, additional options as well as virtualenv, pip setuptools and wheel version for creating powetry venvs

### DISCUSSION THREAD

**Discussion by goodwillcoding** [python-poetry/poetry#9348] [Ideas]
Hi folks,

As far as I am aware there is no way to specify virtualenv script, additional options of versions of pip, setuptools and wheels used for creation for virtualenv. I usually do for reproducibility. Any thoughts on how to go about it? Or am I missing something?

**Comment by finswimmer** [comment]
If your project really rely on specific versions of these tools, I would just add them to the list of dependencies like all other dependencies.

**Reply by goodwillcoding** [reply]
There is no way to do that with virtualenv for one, poetry would not use it one installed inside the virtualenv it creates itself inside the project. As for pip, setuptools and wheel it makes sense to have them before anything from project gets installed. 

As a parallel situation, additional ability to command line options to virtualenv make sense. Or specifying some wrapper script (i.e. virtualenv.sh). For example, I prefer to specify --never-download and --no-periodic updates there. That last one is especially tricky. I do not want to do it on a global level, it's project specific.

**Comment by Secrus** [comment]
If you have a special needs from your virtualenv, you can make it manually activate it and Poetry will pick it up.

**Reply by goodwillcoding** [reply]
I am aware I can and I have done it. But that basically duplicates functionality of poetry, everything from naming virtualenv folders and so on and so on. I've also used PATH modifications and such, but again this would be mostly solved by specifying the path to virtualenv script (or in the case of pyz file a string that include python interpreter). 

On overall this is not uncommon, all the automation software (ansible, saltstack, etc) ... have it as an option because it has been demonstrated to be a real need for reproducibility.

**Reply by goodwillcoding** [reply]
If I were to do a PR with `virtualenv.command =` would it be accepted?

**Reply by Secrus** [reply]
No. That would introduce a lot more complexity than necessary and the envs code is already quite complex. Right now the only thing I can recommend is for you to make your own envs.
